pkg-manager/core/test/catalogs.ts

```diff
@@ -61,9 +63,13 @@ class CatalogTestsController {
     return readYamlFile(path.join(this.workspaceDir, WANTED_LOCKFILE))
   }
 
-  async updateProjectManifest (location: string, manifest: ProjectManifest): Promise<void> {
-    const { selectedProjectsGraph } = await filterPackagesFromDir(this.workspaceDir, [{ parentDir: location }])
-    await selectedProjectsGraph[path.join(this.workspaceDir, location)].package.writeProjectManifest(manifest)
+  updateProjectManifest (location: ProjectId, manifest: ProjectManifest): void {
+    fs.writeFileSync(path.join(this.workspaceDir, location, 'package.json'), JSON.stringify(manifest, null, 2))
```

Comment:
there is no need to write the updated package.json files to fs. core doesn't read these files, it uses solely the objects passed to it.

Reply:
Gotcha. I wanted to make sure these were up to date "just in case", but it's nice to know we can just assume `@pnpm/core` operates off of the in-memory objects entirely.